src/commons/report.py

```diff
@@ -104,25 +109,79 @@ def update_tests_status(input_dict: dict, corio_start_time: datetime, value: dic
     # Reason of the test execution failure.
     test_failed = kwargs.get("test_failed", '')
     test_start_time = corio_start_time + value['start_time']
-    if datetime.now() > test_start_time:
-        input_dict["START_TIME"] = f"Started at {test_start_time.strftime('%Y-%m-%d %H:%M:%S')}"
-        if datetime.now() > (test_start_time + value['min_runtime']):
-            pass_time = (test_start_time + value['min_runtime']).strftime('%Y-%m-%d %H:%M:%S')
-            input_dict["RESULT_UPDATE"] = f"Passed at {pass_time}"
-            total_execution_time = value['min_runtime'] if sequential_run else datetime.now(
-            ) - test_start_time
-        else:
-            # Report In Progress, Fail, Aborted and update status.
-            if input_dict["TEST_ID"] in terminated_tests:
-                LOGGER.error("Test execution terminated due to error in %s.", input_dict["TEST_ID"])
-                input_dict["RESULT_UPDATE"] = "Fail"
-            elif test_failed:
-                input_dict["RESULT_UPDATE"] = "Aborted"
+    start_iterations_count = 0
+    completed_iterations_count = 0
+    fpath = get_test_file_path(corio_start_time, input_dict['TEST_ID'])
+    yaml_file = '/root/corio/workload/master_config.yaml'
+    LOGGER.debug("YAML file selected for parse: %s", yaml_file)
+    yaml_dict = {}
+    with open(yaml_file, "r", encoding="utf-8") as obj:
+        data = yaml.safe_load(obj)
+        yaml_dict.update(data)
+    LOGGER.debug("YAML file data: %s", yaml_dict)
+    if yaml_dict['s3api']['wait_on_operation'] is True:
+        with open(fpath, 'r') as test_log_file:
+            lines = test_log_file.readlines()
+            for line in lines:
+                if 'iteration' in line.lower():
+                    line = line.split()
+                    if 'started' in line:
+                        start_iterations_count+=1
+                    if 'completed' in line:
+                        completed_iterations_count+=1
+            LOGGER.info(start_iterations_count)
+            LOGGER.info(completed_iterations_count)
+        test_log_file.close()
```

Comment:
I think we should improve this code as it will be consuming client memory incase log file getting bigger, so better to use any Linux/Unix command to have expected check. cc: @VikasKumarC

Reply:
will be taken care of in follow up task.